src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```diff
@@ -188,6 +192,25 @@ private boolean mqttVersionHasChanged(Map<String, Object> newNucleusParameters,
         return false;
     }
 
+    private boolean spoolerStorageTypeHasChanged(Map<String, Object> newNucleusParameters,
+                                          DeviceConfiguration currentDeviceConfiguration) {
+        String currentSpoolerStorageType = Coerce.toString(
+                currentDeviceConfiguration.getSpoolerNamespace().findOrDefault(DEFAULT_SPOOL_STORAGE_TYPE,
+                        SPOOL_STORAGE_TYPE_KEY));
+        Map<String, Object> newSpooler = (Map<String, Object>) newNucleusParameters.get(DEVICE_SPOOLER_NAMESPACE);
```

Comment:
do we know that `newNucleusParameters` is non-null at this point?

Reply:
Yes